ASTree.cpp

```diff
@@ -2117,10 +2118,36 @@ PycRef<ASTNode> BuildFromCode(PycRef<PycCode> code, PycModule* mod)
                     PycRef<ASTNode> src = stack.top();
                     stack.pop();
 
-                    if (dest.type() == ASTNode::NODE_MAP) {
-                        dest.cast<ASTMap>()->add(subscr, src);
+                    // If variable annotations are enabled, we'll need to check for them here.
+                    // Python handles a varaible annotation by setting:
+                    // __annotations__['var-name'] = type
+                    bool found_annotated_var = false;
+                    if (variable_annotations) {
+                        if (dest->type() == ASTNode::Type::NODE_NAME) {
+                            if (dest.cast<ASTName>()->name()->isEqual("__annotations__")) {
+                                found_annotated_var = true;
+                            }
+                        }
+                    }
```

Comment:
I would probably put this in a single expression, rather than three nested `if` blocks:

```c++
    const bool found_annotated_var = (variable_annotations && dest->type() == ASTNode::Type::NODE_NAME
                                      && dest.cast<ASTName>()->name()->isEqual("__annotations__"));
```

Reply:
Done! This looks much cleaner now :)